extractor/filesystem/language/swift/swiftutils/podfilelock.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package swiftutils
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+// PodfileLock represents the structure of a Podfile.lock file.
+type PodfileLock struct {
+	Pods            []interface{}     `yaml:"PODS"`
+	SpecChecksums   map[string]string `yaml:"SPEC CHECKSUMS"`
```

Comment:
We only use podfile.Pods from these so let's remove the other fields - otherwise we're loading them into memory (potentially increasing memory usage) for no reason